excise_management/data/ir_model_fields.xml

```diff
@@ -387,4 +387,261 @@ for record in self:
+        <field name="model_id" ref="excise_report"/>
+        <field name="name">x_external_id</field>
+    </record>
+    <record id="document_excise_report" model="ir.model.fields">
+        <field name="ttype">binary</field>
+        <field name="copied" eval="False"/>
+        <field name="field_description">Document</field>
+        <field name="model_id" ref="excise_report"/>
+        <field name="name">x_document</field>
+    </record>
+    <record id="details_computation_excise_report" model="ir.model.fields">
+        <field name="ttype">boolean</field>
+        <field name="field_description">Line Computation</field>
+        <field name="model_id" ref="excise_report"/>
+        <field name="name">x_details_computation</field>
+        <field name="store" eval="False"/>
+        <field name="compute"><![CDATA[
+for record in self:
+    if record.x_state != "done" and record.x_from_date and record.x_to_date:
+        server_action = self.env['ir.actions.server'].browse(self.env.ref('excise_management.ir_action_compute_excise_report_line')).id
+        server_action.with_context(active_ids=[record.id], active_model="x_excise_report").sudo().run()
+        record['x_details_computation'] = True
+        ]]></field>
+    </record>
+    <record id="state_excise_report" model="ir.model.fields">
+        <field name="name">x_state</field>
+        <field name="copied" eval="False"/>
+        <field name="model_id" ref="excise_report" />
+        <field name="ttype">selection</field>
+        <field name="field_description">State</field>
+    </record>
+    <record id="state_excise_report_selection_progress" model="ir.model.fields.selection">
+        <field name="name">In progress</field>
+        <field name="field_id" ref="state_excise_report" />
+        <field name="value">progress</field>
+    </record>
+    <record id="state_excise_report_selection_done" model="ir.model.fields.selection">
+        <field name="name">Done</field>
+        <field name="field_id" ref="state_excise_report" />
+        <field name="value">done</field>
+    </record>
+    <record id="excise_report_line_excise_report" model="ir.model.fields">
+        <field name="ttype">many2one</field>
+        <field name="name">x_excise_report_id</field>
+        <field name="field_description">Excise report ID</field>
+        <field name="model_id" ref="excise_report_line" />
+        <field name="relation">x_excise_report</field>
+        <field name="on_delete">cascade</field>
+    </record>
+    <record id="excise_report_line_currency_id" model="ir.model.fields">
+        <field name="ttype">many2one</field>
+        <field name="field_description">Currency</field>
+        <field name="model_id" ref="excise_report_line"/>
+        <field name="name">x_currency_id</field>
+        <field name="readonly" eval="True"/>
+        <field name="relation">res.currency</field>
+        <field name="compute"><![CDATA[
```

Comment:
Can't we have a related field to company.currency_id?
currently it is compute and store false, so will be calculated every time.